belgif-rest-problem-java-ee/src/main/java/io/github/belgif/rest/problem/ee/jaxrs/JsonParseExceptionMapper.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Smals
+ */
+package io.github.belgif.rest.problem.ee.jaxrs;
+
+import static io.github.belgif.rest.problem.api.InputValidationIssues.invalidStructure;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.io.JsonEOFException;
+
+import io.github.belgif.rest.problem.BadRequestProblem;
+import io.github.belgif.rest.problem.api.InEnum;
+import io.github.belgif.rest.problem.config.ProblemConfig;
+
+@Provider
+public class JsonParseExceptionMapper implements ExceptionMapper<JsonParseException> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JsonParseExceptionMapper.class);
+
+    @Override
+    public Response toResponse(JsonParseException exception) {
+        LOGGER.warn(exception.getMessage(), exception);
+        try {
+            ProblemConfig.setExtIssueTypesEnabled(true);
```

Comment:
We shouldn't alter the setExtIssueTypesEnabled flag here, it is up to the applicaiton owner to enable/disable.

Based on that flag, issue type "urn:problem-type:belgif-ext:input-validation:invalidStructure" (enabled) or "urn:problem-type:belgif:input-validation:invalidInput" (disabled) will be used.